File: agent/src/main/java/com/teamscale/jacoco/agent/GenericExceptionMapper.java
package com.teamscale.jacoco.agent;

import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import javax.ws.rs.ext.ExceptionMapper;

/**
 * Generates a {@link Response} for an exception.
 */
@javax.ws.rs.ext.Provider
public class GenericExceptionMapper implements ExceptionMapper<Throwable> {

	@Override
	public Response toResponse(Throwable e) {
		Response.ResponseBuilder errorResponse = Response.status(Response.Status.INTERNAL_SERVER_ERROR);
		errorResponse.type(MediaType.TEXT_PLAIN_TYPE);
		errorResponse.entity("Message: " + e.getMessage());
		return errorResponse.build();
	}
}

File: agent/src/main/kotlin/com/teamscale/jacoco/agent/GenericExceptionMapper.kt
package com.teamscale.jacoco.agent

import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response
import javax.ws.rs.ext.ExceptionMapper
import javax.ws.rs.ext.Provider

/**
 * Generates a [Response] for an exception.
 */
@Provider
class GenericExceptionMapper : ExceptionMapper<Throwable> {
	override fun toResponse(e: Throwable): Response? =
		Response.status(Response.Status.INTERNAL_SERVER_ERROR).apply {
			type(MediaType.TEXT_PLAIN_TYPE)
			entity("Message: ${e.message}")
		}.build()
}
